frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx

```diff
@@ -1,131 +1,165 @@
+          "error",
+          `Failed to fetch stream statistics: ${returnedError.message}`
+        );
+      });
     });
   });
-  describe("getButton() function", () => {
-    it("if state is null, should return null", () => {
-      const button = wrapper.instance().getButton();
-      expect(button).toBeNull();
+
+  describe("<StreamStaticsCard/> class functions", () => {
+    beforeEach(() => {
+      jest
+        .spyOn(StreamApi, "getStreamStatistics")
+        .mockResolvedValue(dummyStats);
+      wrapper = Enzyme.shallow(<StreamStatisticsCard stream={dummyStream} />);
     });
-    it("if state is not null, should return appropriate button", () => {
-      wrapper.setState(expectedState);
-      const actualButton = wrapper.instance().getButton();
-      expect(actualButton).toStrictEqual(expectedButton);
+    describe("handleStatsChange() function", () => {
+      it("should set the stats state", () => {
+        const randomPreviousState = {
+          stats: new StreamStatisticsInfo(
+            1000,
+            2000,
+            new StreamStatsWindowInfo(12, 13, 14),
+            new StreamStatsLinkInfo(15, 16, 17),
+            new StreamStatsSendInfo(21, 22, 23, 24, 25, 26, 27),
+            new StreamStatsReceiveInfo(31, 32, 33, 34, 35, 36, 37, 38, 39)
+          )
+        };
+        wrapper.instance().setState(randomPreviousState);
+        expect(wrapper.state()).toStrictEqual(randomPreviousState);
+        wrapper.instance().handleStatsChange(dummyStats);
+        expect(wrapper.state()).toStrictEqual(expectedState);
+      });
     });
-  });
-  describe("render() function", () => {
-    it("should render 1 DashboardCard with expected props", () => {
-      wrapper.instance().setState(expectedState);
-
-      const expectedProps = {
-        title: "Statistics",
-        button: expectedButton
-      };
-
-      const dashCard = wrapper.find(DashboardCard);
-      expect(dashCard).toHaveLength(1);
-
-      expect(dashCard.props().title).toBe(expectedProps.title);
-      expect(dashCard.props().button).toStrictEqual(expectedProps.button);
+    describe("getProperties() function", () => {
+      it("it should return a specific array of properties", () => {
+        const actualProperties = wrapper.instance().getProperties();
+        expect(actualProperties).toStrictEqual(expectedProperties);
+      });
     });
-    it("should render 1 Grid component with expected props", () => {
-      expect(wrapper.find(Grid)).toHaveLength(1);
-
-      const expectedProps = { item: true, xs: 12 };
-      const gridProps = wrapper.find(Grid).at(0).props();
-
-      expect(gridProps.item).toBe(expectedProps.item);
-      expect(gridProps.xs).toBe(expectedProps.xs);
+    describe("getButton() function", () => {
+      it("it should return appropriate button", () => {
+        const actualButton = wrapper.instance().getButton();
+        expect(actualButton).toStrictEqual(expectedButton);
+      });
```

Comment:
Did  you remove the test when these two methods return null?

Reply:
I talked to @sharon-ho and apprently the removed tests test for stuff that will never happen so it's all good!